lib/experimental/fonts-api/bc-layer/class-gutenberg-fonts-api-bc-layer.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Fonts API BC Layer helpers.
+ *
+ * BACKPORT NOTE: Do not backport this file to Core.
+ * This file is now part of the API's Backwards-Compatibility (BC) layer.
+ *
+ * @package    Gutenberg
+ * @subpackage Fonts API
+ * @since      15.7.0
+ */
+
+/**
+ * Class Gutenberg_Fonts_API_BC_Layer
+ *
+ * BACKPORT NOTE: Do not backport this file to Core.
+ */
+class Gutenberg_Fonts_API_BC_Layer {
+
+	/**
+	 * Determines if the given fonts array is the deprecated array structure.
+	 *
+	 * @param array $fonts Array of fonts to check.
+	 * @return bool True when deprecated structure, else false.
+	 */
+	public static function is_deprecated_structure( array $fonts ) {
```

Comment:
Good question. The assumption is the array structure will either be the deprecated structure of the new structure, but not a mixture of both. 

@anton-vlasenko Do you foresee a flaw in that assumption, maybe that an array of fonts could contain a mixture?

Reply:
Another way to do this is:

- Iterate through all the first dimension keys.
- First key that is not a non-empty string, return `true`.

Reply:
The more I think about, it probably should check all of the first dimension keys. Done here https://github.com/WordPress/gutenberg/pull/50077/commits/12db6074a9825cc58f7c68f8999428091fcce0df.

Reply:
> Do you foresee a flaw in that assumption, maybe that an array of fonts could contain a mixture?

@hellofromtonya 
At first, I was wondering why only the first element of the array was being checked, and I thought that it might be because the first element was special in some way. :)  
However, now I understand that it is not the case, and the difference lies in the old versus new fonts format.
Thank you for explaining this to me.
I fully support the idea of being more cautious and checking all the array keys.